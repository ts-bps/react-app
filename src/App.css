body {
  font-size: 20;
}
.helloWorld {
  text-align: center;
  padding: 20;
}

.helloWorld .hello {
  padding: 20;
}
.helloWorld .counter .count {
  padding: 20;
}

.helloWorld .counter .increment {
  padding: 20;
}

.helloWorld .counter .decrement {
  padding: 20;
  margin: 10;
}
