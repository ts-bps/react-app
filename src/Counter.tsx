import * as React from "react";

export class Counter extends React.Component<{ count: number }> {
  state = {
    count: 0
  };
  render() {
    return (
      <div className="counter">
        <div className="count">{this.state.count}</div>
        <button
          className="increment"
          onClick={() => {
            this.setState({ count: this.state.count + 1 });
          }}
        >
          ++
        </button>
        <button
          className="decrement"
          onClick={() => {
            this.setState({ count: this.state.count - 1 });
          }}
        >
          --
        </button>
      </div>
    );
  }
}
