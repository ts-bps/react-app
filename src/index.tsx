import * as React from "react";
import * as ReactDOM from "react-dom";

import { HelloWorld } from "./App";
import "./App.css";

ReactDOM.render(<HelloWorld />, document.getElementById("app"));
