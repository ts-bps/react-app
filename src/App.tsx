import * as React from "react";

import { Counter } from "./Counter";

export const Hello: React.StatelessComponent<{ name: string }> = ({ name }) => {
  return <div className="hello">Hello {name}</div>;
};

Hello.defaultProps = {
  name: "world"
};
export const HelloWorld: React.StatelessComponent<{}> = () => (
  <div className="helloWorld">
    <Hello name="world" />
    <Counter />
  </div>
);
