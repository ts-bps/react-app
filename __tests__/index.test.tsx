import * as React from "react";
import * as Renderer from "react-test-renderer";
import { HelloWorld } from "../src/App";
import { Counter } from "../src/Counter";

describe("HelloWorld", () => {
  test("Renders", () => {
    const tree = Renderer.create(<HelloWorld />);
    expect(tree).toMatchSnapshot();
  });
});

describe("Counter", () => {
  test("Renders", () => {
    const tree = Renderer.create(<Counter />);
    expect(tree).toMatchSnapshot();
  });
});
