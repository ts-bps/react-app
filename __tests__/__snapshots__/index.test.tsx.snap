// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Renders 1`] = `
<div
  className="counter"
>
  <div
    className="count"
  >
    0
  </div>
  <button
    className="increment"
    onClick={[Function]}
  >
    ++
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    --
  </button>
</div>
`;

exports[`HelloWorld Renders 1`] = `
<div
  className="helloWorld"
>
  <div
    className="hello"
  >
    Hello 
    world
  </div>
  <div
    className="counter"
  >
    <div
      className="count"
    >
      0
    </div>
    <button
      className="increment"
      onClick={[Function]}
    >
      ++
    </button>
    <button
      className="decrement"
      onClick={[Function]}
    >
      --
    </button>
  </div>
</div>
`;
